### DISCUSSION TITLE
Git initial-branch option ignored

### DISCUSSION THREAD

**Discussion by DoktorNik** [community/community#23592] [New to GitHub]
<p>For  context, I’m still trying to get things setup for git and github to work together well, and I’m getting pretty close.</p>
<p>The problem I’m currently facing is that git uses the master branch, but github uses the main branch. I read a long article about why exactly that is, which was nice, but the recommendation to fix it didn’t help.</p>
<p>The artcle recommended<br>
<code>git init --initial-branch=main</code></p>
<p>However, that returns:<br>
<code>warning: re-init: ignored --initial-branch=main</code></p>
<p>I searched for this error online but I didn’t find anything helpful for me.</p>
<p>Any ideas on how to get git to stop ignoring me? Bad git! :slight_smile:</p>
<p>Thanks in advance!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p><code>git init</code> won’t delete existing branches if you call it again in an existing repository, that’s a good thing. In general, there’s no reason to call <code>git init</code> in an existing repository. :wink:</p>
<p>In your existing repository you can just rename the branch:</p>
<pre><code class="lang-bash">git branch -m master main
</code></pre>
<p>For a <em>new</em> repository that init command should work, or you could set the <a href="https://git-scm.com/docs/git-config#Documentation/git-config.txt-initdefaultBranch"><code>init.defaultBranch</code> option</a> to change what <code>git init</code> does without the parameter.</p>

**Comment by DoktorNik** [comment]
<p>While I disagree that ignoring user input, especially without explaining why, is a good thing, I appreciate the information.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="FizzleFuze" data-post="1" data-topic="182112">
<div class="title">
<div class="quote-controls"></div>
 FizzleFuze:</div>
<blockquote>
<p>warning: re-init</p>
</blockquote>
</aside>
<p>It told you why, though admittedly not very verbosely. :wink:</p>

**Comment by ghost** [comment]
<p>works like a charm thanks!</p>
